google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Job.java

```diff
@@ -410,9 +415,34 @@ public Job reload(JobOption... options) {
    * @throws BigQueryException upon failure
    */
   public boolean cancel() {
+    checkNotDryRun("cancel");
     return bigquery.cancel(getJobId());
   }
 
+  private void checkNotDryRun(String op) {
+    checkNotDryRun(getConfiguration(), op);
+  }
+
+  static void checkNotDryRun(JobConfiguration jobConfig, String op) {
+    QueryJobConfiguration config;
+    if (jobConfig instanceof QueryJobConfiguration) {
+      config = (QueryJobConfiguration) jobConfig;
+    } else {
+      return;
+    }
+
+    // Not "if (config.dryRun())", which fails if the Boolean is null.
+    // The below assumes that null means not-dryrun.
+    if (config.dryRun() == Boolean.TRUE) {
```

Comment:
I just realized; this is incorrect. `Boolean.TRUE` does *not* compare equal to `new Boolean(true)`. I'll fix this